src/blueprint/from_config.rs

```diff
@@ -29,12 +29,15 @@ type Valid<A> = ValidDefault<A, String>;
 pub fn config_blueprint(config: &Config) -> Valid<Blueprint> {
   let output_types = config.output_types();
   let input_types = config.input_types();
-  let schema = to_schema(config)?;
-  let definitions = to_definitions(config, output_types, input_types)?;
-  let server: Server = Server::try_from(config.server.clone())?;
+  let schema = to_schema(config);
+  let definitions = to_definitions(config, output_types, input_types);
+  let server: Valid<Server> = Server::try_from(config.server.clone());
   let upstream = config.upstream.clone();
-  valid_base_url(upstream.base_url.as_ref())?;
-  let blueprint = Blueprint { schema, definitions, server, upstream };
+  let server = valid_base_url(upstream.base_url.as_ref())
+    .validate_or(schema.clone())
+    .validate_or(definitions.clone())
+    .validate_or(server)?;
+  let blueprint = Blueprint { schema: schema?, definitions: definitions?, server, upstream };
```

Comment:
Come to think of it, I think we should use `validate_both` and then a `map` operator to combine and create a blueprint.

```rust
let (schema, definitions, server, upstream) = schema
  .validate_both(definitions)
  .validate_both(server)
  .validate_both(upstream)
  .map(|tuple| tuple.flatten())?
  
```

Checkout
https://crates.io/crates/tuples

Reply:
why need flatten its not necessary computation
let (((_, schema), definitions), server) can destructure